migration/idempotent/007-tracing-tags.sql

```diff
@@ -96,10 +96,57 @@ $fnc$
 $fnc$;
 GRANT EXECUTE ON FUNCTION _ps_trace.tag_v_eq_matching_tags(_tag_key pg_catalog.text, _value pg_catalog.jsonb) TO prom_reader;
 COMMENT ON FUNCTION _ps_trace.tag_v_eq_matching_tags
-IS 'This function is a part of custom _ps_trace.tag_v type which is a wrapper for 
+IS 'This function is a part of custom _ps_trace.tag_v type which is a wrapper for
 the built-in jsonb. The tag_map_rewrite support function, attached to tag_v_eq,
 will use this function instead, if it can.';
 
+CREATE OR REPLACE FUNCTION _ps_trace.text_matches(_value pg_catalog.text)
+    RETURNS pg_catalog.jsonb[]
+    LANGUAGE sql IMMUTABLE
+    -- Note: no explicit `SET SCHEMA` because we want this function to be inlined
+    PARALLEL SAFE AS
+$fnc$
+SELECT CASE
+ WHEN _value IN ('true', 'false') THEN ARRAY[pg_catalog.to_jsonb(_value), pg_catalog.to_jsonb(_value::pg_catalog.bool)]
+ --regex from https://www.regular-expressions.info/floatingpoint.html
+ --the json standard specifies floats not the eqivalent of postgres numerics
+ WHEN _value ~ '^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$' THEN ARRAY[pg_catalog.to_jsonb(_value), pg_catalog.to_jsonb(_value::pg_catalog.float8)]
```

Comment:
Yes, but AFAICT JSONB stores them similar to how numerics are stored and can represent numbers with precision unachievable under float8, but perfectly possible in a JSON string.

Reply:
You are absolutely right. Changed this to use numeric types.